third_party/terraform/utils/config.go.erb

```diff
@@ -711,19 +714,29 @@ func (c *Config) NewServiceUsageClient(userAgent string) *serviceusage.Service {
 
 // staticTokenSource is used to be able to identify static token sources without reflection.
 type staticTokenSource struct {
-	oauth2.TokenSource
+      oauth2.TokenSource
 }
 
 func (c *Config) GetCredentials(clientScopes []string) (googleoauth.Credentials, error) {
+
 	if c.AccessToken != "" {
 		contents, _, err := pathOrContents(c.AccessToken)
 		if err != nil {
 			return googleoauth.Credentials{}, fmt.Errorf("Error loading access token: %s", err)
 		}
+		token := &oauth2.Token{AccessToken: contents}
+
+		if c.ImpersonateServiceAccount != "" {
+			opts := []option.ClientOption{option.WithTokenSource(token), option.ImpersonateCredentials(c.ImpersonateServiceAccount, c.ImpersonateServiceAccountDelegates...), option.WithScopes(clientScopes...)}
```

Comment:
I think we'll want to pass in `oauth2.StaticTokenSource(token)` instead of `token` as `WithTokenSource` needs a TokenSource

Reply:
fixed